database/src/pool/postgres.rs

```diff
@@ -1413,6 +1414,121 @@ where
             .await
             .unwrap();
     }
+
+    async fn get_benchmark_requests_by_status(
+        &self,
+        statuses: &[BenchmarkRequestStatus],
+        days: Option<i32>,
+    ) -> anyhow::Result<Vec<BenchmarkRequest>> {
+        // There is a small period of time where a try commit's parent_sha
+        // could be NULL, this query will filter that out.
+        let mut query = "
+                SELECT
+                    tag,
+                    parent_sha,
+                    pr,
+                    commit_type,
+                    status,
+                    created_at,
+                    completed_at,
+                    backends,
+                    profiles
+                FROM benchmark_request
+                WHERE status = ANY($1)
+                AND (commit_type <> 'try' OR parent_sha IS NOT NULL) "
+            .to_string();
+
+        // Optionally add interval
+        let rows = if let Some(days) = days {
+            // postgres expects the interval to be a `float` so we cast with;
+            // `::INT`, we can't do; `INTERVAL '$2 day'` which, while looking
+            // natural is invalid.
+            query += "AND created_at > current_date - $2::INT * INTERVAL '1 day'";
+            self.conn().query(&query, &[&statuses, &days]).await
+        } else {
+            self.conn().query(&query, &[&statuses]).await
+        }
+        .context("Failed to get benchmark requests")?;
+
+        let benchmark_requests = rows
+            .iter()
+            .map(|row| {
+                let tag = row.get::<_, String>(0);
```

Comment:
Here we allocate `String`, then we pass it to `create_xyz` as `&str`, and inside that function we turn it into a `String` again. I think we can do `row::get::<_, &str>` to avoid the first allocation for each string here.